src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(health = 100, name) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(100, name);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(100, name);
+  }
+
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        const index = Animal.alive.indexOf(target);
+
+        if (index > -1) {
+          Animal.alive.splice(index, 1);
```

Comment:
[CODE PERFOMANCE] - Instead of using a combination of `indexOf` and `splice` to remove the dead animal from the `Animal.alive` array, which involves nesting loops and can be inefficient, prefer using the `filter` method. This will avoid the performance cost of nested loops and is more declarative. You can implement it directly in the `bite` method after checking the health of the target, or you can create a separate method that cleans up dead animals from the `Animal.alive` array.